src/hyperlight_host/src/sandbox/initialized_multi_use.rs

```diff
@@ -121,8 +121,22 @@ impl MultiUseSandbox {
         dispatch_ptr: RawPtr,
         #[cfg(gdb)] dbg_mem_access_fn: Arc<Mutex<SandboxMemoryManager<HostSharedMemory>>>,
     ) -> MultiUseSandbox {
+        let id = SANDBOX_ID_COUNTER.fetch_add(1, Ordering::Relaxed);
+
+        // Register with crash handler if dumps are enabled
+        #[cfg(feature = "crashdump")]
+        if vm.runtime_config().guest_core_dump
```

Comment:
Do we want this on by default?
Imagine running a huge number of sandboxes and something hangs, and you want to terminate it and it starts dumping everything.
I know this is not a production use case, but still. Let me know what you think.
Also, I am curious what happens if we're running 100+ sandboxes of big sizes 500MiB 😄